src/qubex/experiment/mixin/characterization_mixin.py

```
@@ -1505,18 +1558,26 @@ def resonator_spectroscopy(
         self,
         target: str,
         *,
-        frequency_range: ArrayLike = np.arange(9.75, 10.75, 0.002),
+        frequency_range: ArrayLike | None = None,
         power_range: ArrayLike = np.arange(-60, 5, 5),
```

Ruff (0.8.2)

Do not perform function call `np.arange` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)